nibabel/streamlines/tck.py

```diff
@@ -0,0 +1,440 @@
+
+        # Set endianness and _dtype attributes in the header.
+        hdr[Field.ENDIANNESS] = '<'
+        if hdr['datatype'].endswith('BE'):
+            hdr[Field.ENDIANNESS] = '>'
+
+        hdr['_dtype'] = np.dtype(hdr[Field.ENDIANNESS] + 'f4')
+
+        # Keep the file position where the data begin.
+        hdr['_offset_data'] = int(hdr['file'].split()[1])
+
+        # Set the file position where it was, if it was previously open.
+        if start_position is not None:
+            fileobj.seek(start_position, os.SEEK_SET)
+
+        return hdr
+
+    @staticmethod
+    def _read(fileobj, header, buffer_size=4):
+        """ Return generator that reads TCK data from `fileobj` given `header`
+
+        Parameters
+        ----------
+        fileobj : string or file-like object
+            If string, a filename; otherwise an open file-like object
+            pointing to TCK file (and ready to read from the beginning
+            of the TCK header). Note that calling this function
+            does not change the file position.
+        header : dict
+            Metadata associated with this tractogram file.
+        buffer_size : float, optional
+            Size (in Mb) for buffering.
+
+        Yields
+        ------
+        points : ndarray of shape (n_pts, 3)
+            Streamline points
+        """
+        dtype = header["_dtype"]
+        coordinate_size = 3 * dtype.itemsize
+        # Make buffer_size an integer and a multiple of coordinate_size.
+        buffer_size = int(buffer_size * MEGABYTE)
+        buffer_size += coordinate_size - (buffer_size % coordinate_size)
+
+        with Opener(fileobj) as f:
+            start_position = f.tell()
+
+            # Set the file position at the beginning of the data.
+            f.seek(header["_offset_data"], os.SEEK_SET)
+
+            eof = False
+            buff = b""
+            pts = []
+
+            i = 0
+
+            while not eof or not np.all(np.isinf(pts)):
+
+                if not eof:
+                    bytes_read = f.read(buffer_size)
+                    buff += bytes_read
+                    eof = len(bytes_read) == 0
+
+                # Read floats.
+                pts = np.frombuffer(buff, dtype=dtype)
+
+                # Convert data to little-endian if needed.
+                if dtype != '<f4':
+                    pts = pts.astype('<f4')
+
+                pts = pts.reshape([-1, 3])
+                idx_nan = np.arange(len(pts))[np.isnan(pts[:, 0])]
+
+                # Make sure we've read enough to find a streamline delimiter.
+                if len(idx_nan) == 0:
+                    # If we've read the whole file, then fail.
+                    if eof and not np.all(np.isinf(pts)):
+                        msg = ("Cannot find a streamline delimiter. This file"
+                               " might be corrupted.")
+                        raise DataError(msg)
+
+                    # Otherwise read a bit more.
```

Comment:
No I don't think you missed anything - I see now that we'll go back to the while condition, then exit the loop.